typing/ctype.ml

```diff
@@ -3048,8 +3048,8 @@ and unify_row_field env fixed1 fixed2 rm1 rm2 l f1 f2 =
         begin match tl1 @ tl2 with [] -> false
         | t1 :: tl ->
             if c1 || c2 then raise_unexplained_for Unify;
-            List.iter (unify env t1) tl;
-            !e1 <> None || !e2 <> None
+            if List.for_all (unify_eq t1) tl then false else
+            (List.iter (unify env t1) tl; true)
```

Comment:
Note the infinite loop in #10762 stems from this change: the `unify_eq` tests fails because the conjunctions contains a type abbreviation which is only expanded in `unify`.